Acme/Certificate/Extractor/CertificateExtractor.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the ACME PHP library.
+ *
+ * (c) Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace AcmePhp\Bundle\Acme\Certificate\Extractor;
+
+use AcmePhp\Bundle\Exception\ParsingCertificateException;
+use AcmePhp\Core\Ssl\Certificate;
+
+/**
+ * Extract data from certificate file.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class CertificateExtractor implements ExtractorInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'cert.pem';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function extract($content)
+    {
+        $rawData = openssl_x509_parse($content);
+        if (!$rawData) {
+            throw new ParsingCertificateException(
+                sprintf('Certificate parsing failed with error: %s', openssl_error_string())
+            );
+        }
+
+        return [
```

Comment:
What do you think of creating a read only value object for that structure? It seems logical and let ide autocomplete